gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/CseValidNetworkShifter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.cse_valid.app;
+
+import com.farao_community.farao.cse_valid.app.exception.CseValidShiftFailureException;
+import com.farao_community.farao.dichotomy.api.NetworkShifter;
+import com.farao_community.farao.dichotomy.api.exceptions.GlskLimitationException;
+import com.farao_community.farao.dichotomy.api.exceptions.ShiftingException;
+import com.farao_community.farao.dichotomy.shift.LinearScaler;
+import com.farao_community.farao.dichotomy.shift.SplittingFactors;
+import com.powsybl.glsk.api.GlskDocument;
+import com.powsybl.iidm.network.Network;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Oualid Aloui {@literal <oualid.aloui at rte-france.com>}
+ */
+
+@Component
+public class CseValidNetworkShifter {
+
+    private static final double SHIFT_TOLERANCE = 1;
+
+    private final FileImporter fileImporter;
+
+    public CseValidNetworkShifter(FileImporter fileImporter) {
+        this.fileImporter = fileImporter;
+    }
+
+    public NetworkShifter getNetworkShifterWithSplittingFactors(TTimestampWrapper timestampWrapper,
+                                                                Network network,
+                                                                String glskUrl) {
+        GlskDocument glskDocument = fileImporter.importGlsk(glskUrl);
+        return new LinearScaler(
+                glskDocument.getZonalScalable(network),
+                new SplittingFactors(timestampWrapper.getImportCornerSplittingFactors()),
+                SHIFT_TOLERANCE);
+    }
+
+    public NetworkShifter getNetworkShifterReduceToFranceAndItaly(TTimestampWrapper timestampWrapper,
+                                                                  Network network,
+                                                                  String glskUrl) {
+        GlskDocument glskDocument = fileImporter.importGlsk(glskUrl);
+        return new LinearScaler(
+                glskDocument.getZonalScalable(network),
+                new SplittingFactors(timestampWrapper.getExportCornerSplittingFactorsMapReduceToFranceAndItaly()),
+                SHIFT_TOLERANCE);
+    }
+
+    NetworkShifter getNetworkShifterWithShiftingFactors(TTimestampWrapper timestampWrapper,
+                                                        Network network,
+                                                        String glskUrl) {
+        GlskDocument glskDocument = fileImporter.importGlsk(glskUrl);
+        return new LinearScaler(
+                glskDocument.getZonalScalable(network),
+                new SplittingFactors(timestampWrapper.getExportCornerSplittingFactors()),
+                SHIFT_TOLERANCE);
+    }
```

Comment:
I'm not convinced these are very useful, the only thing that change is the map inside splitting factors object

Reply:
At least factorize your methods : 
```java
    public NetworkShifter getNetworkShifterWithSplittingFactors(TTimestampWrapper timestampWrapper,
                                                                Network network,
                                                                String glskUrl) {
        return getNetworkShifter(timestampWrapper.getImportCornerSplittingFactors(), network, glskUrl);
    }

    public NetworkShifter getNetworkShifterReduceToFranceAndItaly(TTimestampWrapper timestampWrapper,
                                                                  Network network,
                                                                  String glskUrl) {
        return getNetworkShifter(timestampWrapper.getExportCornerSplittingFactorsMapReduceToFranceAndItaly(), network, glskUrl);
    }

    NetworkShifter getNetworkShifterWithShiftingFactors(TTimestampWrapper timestampWrapper,
                                                        Network network,
                                                        String glskUrl) {
        return getNetworkShifter(timestampWrapper.getExportCornerSplittingFactors(), network, glskUrl);
    }

    NetworkShifter getNetworkShifter(Map<String, Double> splittingFactors, Network network, String glskUrl) {
        GlskDocument glskDocument = fileImporter.importGlsk(glskUrl);
        return new LinearScaler(
                glskDocument.getZonalScalable(network),
                new SplittingFactors(splittingFactors),
                SHIFT_TOLERANCE);
    }
```